pyproject.toml

```diff
@@ -121,6 +122,19 @@ legacy_tox_ini = """
     setenv =
         DJANGO_SETTINGS_MODULE = test_app.sqlite3settings
 
+    [docker:db]
+    image = postgres:15
+    environment =
+        POSTGRES_PASSWORD=dabing
+        POSTGRES_USER=dab
+        POSTGRES_DB=dab_db
+    healthcheck_cmd = pg_isready -U dab -d dab_db
+    healthcheck_interval = 10
+    healthcheck_timeout = 5
+    healthcheck_retries = 5
+    expose =
+        DB_PORT=5432/tcp
```

Comment:
I would really rather not _duplicate_ the parameters. Even the port doesn't seem to be completely the same as

https://github.com/ansible/django-ansible-base/blob/devel/docker-compose.yml

But looking into tox-docker library, unfortunately, it looks like this may not be possible.

Re-phrasing, I would rather this _not_ list things like `image` and `environment`, but instead, reference the name of the service and the docker-compose.yml file.

https://github.com/tox-dev/tox-docker

Similar to:

```ini
[docker:app]
dockerfile = {toxinidir}/Dockerfile
```

but instead of Dockerfile, I'd like it to reference the docker-compose file, so we are only maintaining 1 thing.

Reply:
Yeah I had the same thought.  This is really an entirely different DB though, that only exists during the tox run (unless the flag is added to hold it for debugging) and for different purposes so I wasn't sure it is a big deal.  I could make the values totally different so it doesn't seem like duplication. :)

The port difference is because tox-docker handles it automatically, allowing parallel tox execution by using a different port for each instance.  It sets DB_PORT for the test session to whatever it chose.  But not sure if this is valuable for us (doubt it).

I also looked into the tool docs as well and one method I found that could consolidate them would be to use an environment file called .devdata.env which is loaded by tox-envfile plugin and can be also loaded by docker compose.  As you found, tox-docker does not support docker compose specs.

I suppose another method could be to extract a dockerfile and use the dockerfile directive in docker compose for the db container (and also use it in tox).  There we would lose the auto-port function, but I'm not sure if we would ever run parallel tox sessions on the same host anyway.

Reply:
Hi @AlanCoding Please take a look now.  I used the common Dockerfile approach and that seems to work well for both tools.  If tox-docker ever decides to support docker-compose then it can be moved back into the docker-compose.yml.